docs/docs/routing.md

```diff
@@ -0,0 +1,241 @@
+---
+
+Routing is the process of choosing what to render based on the browser's current URL.
+
+The simplest way to implement Routing in a React application is with a top-level Component that chooses which content to render by URL. For example, a top-level `App` component may decide what to render based on [window.location.pathname](https://developer.mozilla.org/en-US/docs/Web/API/Location):
+
+```js
+function Menu() {
+  return (
+    <div>
+      <a href="/news">News</a>
+      <a href="/profile">Profile</a>
+    </div>
+  );
+}
+
+class App extends React.Component {
+  render() {
+    const location = window.location;
+  
+    let content;
+    if (location.pathname == '/news') {
+      content = <h1>News</h1>;
+    }
+    else if (location.pathname == '/profile') {
+      content = <h1>Profile</h1>;
+    }
+    else {
+      content = <h1>Not Found</h1>;
+    }
+    
+    return (
+      <div>
+        <Menu />
+        <div>{content}</div>
+      </div>
+    );
+  }
+}
+```
+
+But while this app will work, it has one major issue; it will reload the entire page every time a user clicks a link.
+
+## HTML5 History
+
+The HTML5 [History API](https://developer.mozilla.org/en/docs/Web/API/History) makes it possible to update the browser's URL without causing a page reload. It also emits browser events when the user clicks *Back* or *Forward*. By using this API, you can write applications that respond faster and behave as expected when *Back* or *Forward* are clicked.
+
+Adding support for HTML5 History involves three steps:
+
+1. Exporting a history wrapper object
+2. Ensuring links call the History API instead of reloading the page
+3. Updating your rendered content when the current URL changes
+
+## History wrapper objects
+
+The HTML5 History API is not without issues.
+
+Firstly, there are inconsistencies between how the different browsers emit navigation events. Notably, some versions of Chrome and Safari will emit an event on page load while other versions won't.
+
+Secondly, navigation events are only emitted when they're caused by external stimuli such as the *Back* or *Forward* buttons. When the application itself updates the URL, nothing will happen. This means that the application will need to manually pass the new URL from the link that caused it to the top-level component that is handling routing.
+
+Typically, these issues are solved by creating a wrapper around the HTML5 History API. This wrapper will do two things:
+
+- It will normalise events between browsers
+- It will emit navigation events caused by both the application and external stimuli
+
+While it is certainly possible to create your own wrapper, most React applications in the wild make use of the [history](https://github.com/mjackson/history) package.
+
+```bash
+npm install history --save
+```
+
+As the History API is global, the wrapper will also need to be accessible everywhere. A good place to put it is in a `historyWrapper.js` file at the root of your source tree.
```

Comment:
I want to avoid "wrapper" terminology in this doc, it just makes everything sound more complicated than it is. I would also avoid recommending any particular filename / location suggestions. In three months you'll see a cargo cult of history wrappers being placed in this location, and pointless discussions about it being a Best Practice. 😄